guides/github-actions.md

```diff
@@ -0,0 +1,75 @@
+# GitHub Actions recipes
+
+We use [GitHub Actions](https://github.com/features/actions) for some of our contionuous integration and other automations.
+
+This guide provides some common step recipes that may be useful in custom actions. We also have a collection of common [workflow templates](https://github.com/cfpb/.github) for CI and release that make use of these steps. 
+
+## Workflow templates
+
+Our [workflow templates](https://github.com/cfpb/.github) provide common recipes for consumerfinance.gov-specific apps as well as our more general-purpose libraries.
+
+To use our workflow templates, click on the "Actions" tab in a CFPB GitHub repository. If the repository does not already have a GitHub Actions workflow, you will be presented with the option of creating one from one of our templates:
+
+![GitHub Actions template listing](github-actions-templates.png)
+
+If the repository already has a GitHub Actions workflow, click on the "New workflow" button at the top of the workflow list on the left. 
+
+![GitHub Actions New workflow button](github-actions-new-workflow.png)
+
+Then you will be presented with the option to create one from one of our templates, shown above.
+
+New workflow templates can be created in [github.com/cfpb/.github](https://github.com/cfpb/.github) based on [GitHub's workflow template documentation](https://docs.github.com/en/actions/configuring-and-managing-workflows/sharing-workflow-templates-within-your-organization).
+
+## Common step recipes
+
+### Fetching git history for setuptools-git-version
+
+Some of our consumerfinance.gov satellite apps use [setuptools-git-version](https://github.com/pyfidelity/setuptools-git-version) to set the Python package version from the latest git tag. This requires tags to be available in the checkout in the GitHub action: 
+
+```yml
+steps:
+  - uses: actions/checkout@v2
+
+  - name: Fetch tags and commits needed for setuptools-git-version
+    run: |
+      git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+      git fetch origin ${{ github.head_ref }} && git checkout ${{ github.head_ref }}
+```
+
+The intention is for the command `git describe --tags --long --dirty` to succeed.
+
+## Attaching a Python wheel file to a GitHub release
+
+Some of our consumerfinance.gov satellite apps have a Python wheel package file attached to their GitHub releases. We do it by outputting the wheel file name from the build step and then reading it in the upload step:
+
+```yml
+steps:
+  - name: Build the Python packages
+    id: build
+    run: |
+        python setup.py sdist bdist_wheel
```

Comment:
We don't need the sdist for our satellite apps. Do you think there's some value still in creating and uploading it? If not, here are a few code suggestions to remove that.

```suggestion
        python setup.py bdist_wheel
```

Reply:
@willbarton correct me if I'm wrong, but I believe we're able to build `none-any` wheels for all of our existing packages because they're all pure Python. It's at least conceivably possible that in future we might have some that don't meet this criteria, which would mean we'd have to build platform-specific wheels, and thus would want to distribute the source.

If there's no real downside to also building and distributing the source I'd vote that we keep doing so.

Reply:
I think that's true, and I'd agree — there's no downside to including the sdist.

Reply:
:+1: